docs/designs/010-metadata-retrieval.md

```diff
@@ -0,0 +1,255 @@
+
+```
+    ┌────────────────────────────────────────────────────────┐
+    │  GPU Metadata Collector DaemonSet (per node)           │
+    │                                                        │
+    │  Init Container:                                       │
+    │  - Uses go-nvml to collect GPU metadata                │
+    │  - Writes to /var/lib/nvsentinel/                      │
+    │    - gpu_metadata.json (all GPU info + PCI mapping)    │
+    │  - Exits after writing files                           │
+    │                                                        │
+    │  Main Container: pause (minimal resource usage)        │
+    └────────────────────────────────────────────────────────┘
+                              │
+                              │ (writes to hostPath)
+                              ▼
+                    ┌─────────────────────┐
+                    │  Host Path Mount    │
+                    │                     │
+                    │  /var/lib/nvsentinel│
+                    │                     │
+                    └─────────────────────┘
+                              │
+                              │ (mounted read-only)
+                ┌─────────────┴─────────────┐
+                │                           │
+                ▼                           ▼
+    ┌─────────────────────┐     ┌─────────────────────┐
+    │ Syslog Health       │     │ GPU Health Monitor  │
+    │ Monitor             │     │ (Python)            │
+    │ (Go)                │     │                     │
+    │                     │     │                     │
+    │ Input: dmesg/syslog │     │ Input: DCGM         │
+    │ (has PCI ID)        │     │                     │
+    │                     │     │                     │
+    │ Reads from:         │     │ Reads from:         │
+    │ /var/lib/nvsentinel/│     │ /var/lib/nvsentinel/│
+    │                     │     │                     │
+    │ Enriches with:      │     │ Enriches with:      │
+    │ - GPU UUID          │     │ - GPU UUID          │
+    │   (lookup via PCI)  │     │ - PCI addresses     │
+    │ - Chassis serial    │     │ - Chassis serial    │
+    └─────────────────────┘     └─────────────────────┘
+                │                           │
+                │ (health events)           │ (health events)
+                └─────────────┬─────────────┘
+                              │
+                              ▼
+                    ┌─────────────────────┐
+                    │ Platform Connector  │
+                    │                     │
+                    │ Augments with:      │
+                    │ - Other labels      │◄────────┐
+                    │ - Network topology  │         │
+                    │ - Provider ID       │         │
+                    └─────────────────────┘         │
+                                                    │
+                                          ┌─────────────────────┐
```

Comment:
The arrow should be here, i.e., the platform connector reach out to the kube api server and not the other way around